src/sql/src/plan/statement/ddl.rs

```diff
@@ -712,8 +712,21 @@ pub fn plan_create_view(
     } else {
         scx.allocate_name(normalize::object_name(name.to_owned())?)
     };
+    let (mut relation_expr, mut desc, finishing) =
+        query::plan_root_query(scx, query.clone(), QueryLifetime::Static)?;
+    relation_expr.bind_parameters(&params)?;
+    //TODO: materialize#724 - persist finishing information with the view?
+    relation_expr.finish(finishing);
+    let relation_expr = relation_expr.decorrelate();
     let replace = if *if_exists == IfExistsBehavior::Replace {
         if let Ok(item) = scx.catalog.resolve_item(&name.clone().into()) {
+            if relation_expr.global_uses().contains(&item.id()) {
+                bail!(
+                    "cannot replace {0}view {1}: depended upon by new {1} definition",
+                    if *materialized { "materialized " } else { "" },
```

Comment:
I would actually drop this bit of logic as you can call `CREATE OR REPLACE MATERIALIZED VIEW v` on a `v` that is not actually materialized, which will make this error message a little confusing. Arguably we should reject `CREATE OR REPLACE MATERIALIZED VIEW v` and force everyone through `CREATE OR REPLACE VIEW`, but possibly there are weird backwards compatibility implications there... not sure.